samtranslator/plugins/api/implicit_api_plugin.py

```diff
@@ -500,4 +500,6 @@ def _update_resource_attributes_from_api_event(  # noqa: PLR0913
         api_dict_deletion.add(deletion_policy)
 
         api_dict_update_replace = self.api_update_replace_policies.setdefault(api_id, set())
+        if update_replace_policy not in ["Retain", "Snapshot", "Delete", None]:
```

Comment:
whats the usecase where people would set it to `None`?

Reply:
I'll see if I can remove that. I was thinking it might be necessary if update_replace_policy didn't exist.

Reply:
Then 
```python
if update_replace_policy and update_replace_policy not in ["Retain", "Snapshot", "Delete"]:
```
it sounds more idiomatic

Reply:
And also do we use those values "Retain", "Snapshot" etc? If so, let's make them enum or literal constants

Reply:
Also I vaguely remember that it’s possible to have intrinsic in DeletionPolicy, not sure if it’s the same for UpdateDeplacePolicy?

Reply:
If it isn't defined then it is None. So None has to be there.

Reply:
> Also I vaguely remember that it’s possible to have intrinsic in DeletionPolicy, not sure if it’s the same for UpdateDeplacePolicy?

Yeah I just checked. Some intrinsics seems supported as well as some pseudo-parameters.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/function-refs-in-policy-attributes.html